### PR TITLE
CAPZ Managed Kubernetes evolution proposal

### PR DESCRIPTION
**Pull request by jackfrancis**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

This PR attempts to document considerations for evolving the CAPZ Managed Kubernetes implementation in response to https://github.com/kubernetes-sigs/cluster-api/pull/6988.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: docs/proposals/20221019-capi-managed-cluster-proposal-capz.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+A defensible conclusion from [this Cluster API proposal doc](https://github.com/kubernetes-sigs/cluster-api/pull/6988) is that consistency is not achievable given the current state of Cluster API. As described above, the proposal does describe two acceptable approaches, but that is by definition not consistent. Of all the explanations for the current state of Cluster API Managed Kubernetes inconsistency, the most fundamental one is the absence of a Cluster API Managed Cluster "primitive". So long as there isn't a first class Managed Kubernetes resource in the Cluster API itself, providers will struggle to build Managed Kubernetes solutions that are consistent across the Cluster API ecosystem.
+
+## Conclusions
+
+Achieving durable, consistent Managed Kubernetes interfaces for the entire Cluster API provider community is our highest priority. Doing that work is necessarily a large investment in engineering time + resources, and will involve some short-term inconvenience for existing customers. If we wish to embark on that work with the greatest chance for long-term success we will want to do that work in Cluster API itself, and not across the provider community only.
```
Comment (pydctw) [inline docs/proposals/20221019-capi-managed-cluster-proposal-capz.md:181]: 100% on this.

File: docs/proposals/20221019-capi-managed-cluster-proposal-capz.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+It differs from the proposal in the following ways:
+
+- `AzureManagedCluster` does not exclusively define "cluster"-specific properties (the examples above are networking and IAM)
+- `AzureManagedControlPlane` does not exclusively define "control plane"-specific properties; in fact it defines the entire set of "cluster"-wide configuration supported by CAPZ on top of AKS
+
+In order to better adhere to the recommended implementation model CAPZ would move the bulk of its configuration surface area from `AzureManagedControlPlane` to `AzureManagedCluster`.
+
+## AKS-CAPI API Affinity Observations
+
+As stated above, Cluster API defines a "cluster" as distinct from a "control plane". AKS, however, does not declare such a clean boundary in its own API. Using the `az aks` CLI as an example, we can see that the only current definite abstraction boundary at present (as distinct from the cluster itself is "nodepool". E.g.:
```
Comment (pydctw) [inline docs/proposals/20221019-capi-managed-cluster-proposal-capz.md:105]: Thanks for explaining it with examples. When we wrote the initial managed k8s proposal, we also had difficulties placing some fields into two separate CRDs as it was not very clear whether they belonged to control plane or cluster. Great to understand the perspective from AKS architecture better.

File: docs/proposals/20221019-capi-managed-cluster-proposal-capz.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+---
+
+# Summary of Accepted Managed Kubernetes in CAPI Proposal
+
+The following is a lightly edited copy/paste of the preferred option for Managed Cluster implementions in Cluster API as defined in [this proposal doc](https://github.com/kubernetes-sigs/cluster-api/pull/6988).
+
+Two new resource kinds will be introduced:
+
+ - **<Provider>ManagedControlPlane**: this presents a provider's actual managed control plane. Its spec would only contain properties that are specific to the provisioning & management of a provider's cluster (excluding worker nodes). It would not contain any properties related to a provider's general operating infrastructure, like the networking or project.
+ - **<Provider>ManagedCluster**: this presents the properties needed to provision and manage a provider's general operating infrastructure for the cluster (i.e project, networking, IAM). It would contain similar properties to **<Provider>Cluster** and its reconciliation would be very similar.
```
[the remark this answers is not in the document]
Comment (jackfrancis) [reply docs/proposals/20221019-capi-managed-cluster-proposal-capz.md:21]: Th capz document in this PR is meant to communicate to the capz managed k8s community why or why not we will implement the option #3, or keep the existing option #2, in this already merged proposal:

https://github.com/kubernetes-sigs/cluster-api/pull/6988

@richardcase and @pydctw are doing a similar evaluation for how to evolve managed k8s on capa in the near term.

This document does *not* speak to the longer term effort (in this issue https://github.com/kubernetes-sigs/cluster-api/issues/7494) to standardize managed k8s in *capi itself*. We will work that out as a larger capi community, on a longer time scale.

Hope that clears it up a bit!

**Comment by jackfrancis** [comment]
@CecileRobertMichon @nojnhuh @dtzar I know I suggested we might just close this PR during yesterday's office hours, but after a quick re-read I think it captures the CAPZ perspective that contributed to the creation of a Cluster API "Feature Group" to solve this in capi itself. I've updated the doc conclusions to incorporate that outcome, and I think it would be beneficial for the historical record to merge this PR.

wdyt?

**Comment by CecileRobertMichon** [comment]
re-reading this proposal it's unclear to me what the goals/non-goals and concrete implementation proposed are. It's a great summary of the state of the world but I don't know if it fits the "proposal" part of the CAEP process.

**Comment by jackfrancis** [comment]
@CecileRobertMichon Fair enough, we can keep it via's github seemingly indefinite data store :)

**Pull request closed without merging**